lib/internal/bootstrap/node.js

```diff
@@ -467,16 +467,28 @@
       emitAfter
     } = NativeModule.require('internal/async_hooks');
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, fromPromise) {
+      if (fromPromise) {
+        let env = process.env.NODE_UNHANDLED_REJECTION;
+        if (env) {
+          env = env.toUpperCase();
+          if (env === 'WARN' || env === 'SILENT')
+            return true;
+        } else {
+          // Do not crash by default.
+          return true;
```

Comment:
Why is it better to reach the `_uncaughtException` hook and then not crash here rather than not enter it based on the environment variable?

Reply:
I am not sure what you mean with `reaching` the `uncaughtException` hook. This function is the internal `fatalException` handler. I guess you ask why I do not check from the c++ layer if the environment variable is set to prevent calling this? It was just easier for me to check it here instead of in c++ (since the environment variable could be changed during runtime we have to check the state when we actually want to trigger an uncaught exception - otherwise it would be better to do this in c++ to prevent the extra tracking overhead for `SILENT`).

Reply:
You understood me correctly - I just think the code looks a little weird now - maybe just splitting `uncaughtException`.

Reply:
I rework this again somewhat.